config/samples/dataplane_v1beta1_openstackdataplanenodeset_baremetal_with_ipam.yaml

```diff
@@ -0,0 +1,83 @@
+---
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneNodeSet
+metadata:
+  name: openstack-edpm
+spec:
+  deployStrategy:
+    deploy: true
+  baremetalSetTemplate:
+    deploymentSSHSecret: dataplane-ansible-ssh-private-key-secret
+    bmhLabelSelector:
+      app: openstack
+    ctlplaneInterface: enp1s0
+    dnsSearchDomains:
+      - osptest.openstack.org
+    managementNetwork: CtlPlane
+    baremetalHosts: 2
+  nodes: # This will be NodeSection instead of NodeSpec
+    env:
+      - name: ANSIBLE_FORCE_COLOR
+        value: "True"
+      - name: ANSIBLE_ENABLE_TASK_DEBUGGER
+        value: "True"
+      - name: ANSIBLE_VERBOSITY
+        value: "2"
+    # Defining pre-provisioned servers that don't require provisioning
+    edpm-compute-0:
+      ansibleHost: 172.20.12.67
+      ansibleSSHPrivateKeySecret: dataplane-ansible-ssh-private-key-secret
+      ansibleUser: root
+      # Ansible variables here override ansibleVars defined at the top level.
+      # This allows for node specific overrides.
+      ansibleVars:
+        ctlplane_ip: 172.20.12.67
+        internal_api_ip: 172.17.0.101
+        storage_ip: 172.18.0.101
+        tenant_ip: 172.10.0.101
+        external_ip: 172.20.12.76
+        fqdn_internal_api: edpm-compute-1.example.com
+        ansible_ssh_transfer_method: scp
+    edpm-compute-1:
+      hostName: edpm-compute-1
+      preProvisioned: true
+      # If we need to provision nodes, we can do so via the baremetalSetTemplate
+      networks:
+       - name: CtlPlane
+         subnetName: subnet1
+         defaultroute: true
+       - name: InternalApi
+         subnetName: subnet1
+       - name: Storage
+         subnetName: subnet1
+       - name: Tenant
+         subnetName: subnet1
```

Comment:
expressing the netork congi like this is a much better approch and more scaleable. then doing it with ansible vars